File: crates/qsae-python/examples/basic.py
import qsae

# Basic compression
data = b"Hello, world! This is QSAE Python test. " * 1000

# Compress with default parameters
compressed = qsae.compress(data)
print(f"Original: {len(data)} bytes")
print(f"Compressed: {len(compressed)} bytes")
print(f"Ratio: {len(data) / len(compressed):.2f}:1")

# Decompress
original = qsae.decompress(compressed)
assert original == data
print("Roundtrip successful!")

# Analyze
analysis = qsae.analyze(data)
print(f"Blocks: {len(analysis.entropy_profile)}")
print(f"Switch points: {len(analysis.switch_points)}")
print(f"Avg entropy: {sum(analysis.entropy_profile) / len(analysis.entropy_profile):.2f}")

File: crates/qsae-node/examples/basic.js
const qsae = require('qsae-node');

async function main() {
  // Basic compression
  const data = Buffer.from('Hello, world! '.repeat(1000));

  console.log('Compressing...');
  const result = await qsae.compress(data);

  console.log(`Original: ${result.original_size} bytes`);
  console.log(`Compressed: ${result.compressed_size} bytes`);
  console.log(`Ratio: ${result.ratio.toFixed(2)}:1`);
  console.log(`Duration: ${result.duration_ms}ms`);

  // Decompress
  const original = await qsae.decompress(result.data);
  console.log(`Decompressed: ${original.length} bytes`);

  // Verify roundtrip
  if (Buffer.compare(data, original) === 0) {
    console.log('✓ Roundtrip successful!');
  }

  // Analyze
  const analysis = await qsae.analyze(data);
  console.log(`
Analysis:`);
  console.log(`  Blocks: ${analysis.block_count}`);
  console.log(`  Switch points: ${analysis.switch_points.length}`);
  console.log(`  Avg entropy: ${(analysis.entropy_profile.reduce((a, b) => a + b, 0) / analysis.entropy_profile.length).toFixed(2)}`);
}

main().catch(console.error);
